includes/payment-methods/class-wc-stripe-upe-payment-method.php

```diff
@@ -112,10 +112,11 @@ public function __construct() {
 		$main_settings     = get_option( 'woocommerce_stripe_settings' );
 		$is_stripe_enabled = ! empty( $main_settings['enabled'] ) && 'yes' === $main_settings['enabled'];
 
-		$this->enabled  = $is_stripe_enabled && in_array( static::STRIPE_ID, $this->get_option( 'upe_checkout_experience_accepted_payments', [ 'card' ] ), true ) ? 'yes' : 'no';
-		$this->id       = WC_Gateway_Stripe::ID . '_' . static::STRIPE_ID;
-		$this->testmode = ! empty( $main_settings['testmode'] ) && 'yes' === $main_settings['testmode'];
-		$this->supports = [ 'products', 'refunds' ];
+		$this->enabled    = $is_stripe_enabled && in_array( static::STRIPE_ID, $this->get_option( 'upe_checkout_experience_accepted_payments', [ 'card' ] ), true ) ? 'yes' : 'no';
+		$this->id         = WC_Gateway_Stripe::ID . '_' . static::STRIPE_ID;
+		$this->has_fields = true;
```

Comment:
This is set to fix an issue where a payment method section is not loaded (except for the title) if the description is empty (check the logic [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/templates/checkout/payment-method.php#L28)). As the section is not loaded, the Stripe payment element fails to mount.

Reply:
👍 confirmed that without this, UPE methods (like CashApp) won't display anything when selected, if it doesn't have a description. eg:

<p align="center">
<img width="461" alt="Screenshot 2024-07-29 at 4 12 27 PM" src="https://github.com/user-attachments/assets/43d0a6bb-5521-4628-bcf6-92ae6a08c2a1">
</p>